client/my-sites/themes/theme-showcase.jsx

```diff
@@ -118,17 +118,21 @@ const ThemeShowcase = React.createClass( {
 		trackClick( 'upload theme' );
 	},
 
+	showUploadButton() {
+		const { siteSlug, isMultisite, isJetpack } = this.props;
+
+		return config.isEnabled( 'manage/themes/upload' ) && siteSlug && ! isMultisite &&
+			( isJetpack || config.isEnabled( 'automated-transfer' ) );
```

Comment:
Alternatively, you can do it as a `const` inside `render()`, meaning you get a nice `{ showUploadButton && <Button...` without needing `this.showUp...`

```js
const showUploadButton = (
    config.isEnabled( 'manage/themes/upload' ) &&
    siteSlug &&
    ! isMultisite &&
    ( isJetpack || config.isEnabled( 'automated-transfer' ) )
);
```